### DISCUSSION TITLE
methods.watch isn´t working inside FormProvider

### DISCUSSION THREAD

**Discussion by caiofelixuc** [react-hook-form/react-hook-form#7383] [General] [answered]
I recently discovered react hook form , and i am trying to make a Form component that its currently taking a single input as children.
Form component code:
![image](https://user-images.githubusercontent.com/92024680/147581777-151901fc-95b8-48a4-afe4-79581660490c.png)
Input component code: 
![image](https://user-images.githubusercontent.com/92024680/147581952-3f3e6d6f-6ed9-41f1-ae68-a9015f6b8a17.png)

I am trying to  access the input value on every change, using the watch method( do not know if this is correct ), but it always return blank (same for getValues).
![image](https://user-images.githubusercontent.com/92024680/147582099-67874f6d-cefb-4a79-8591-57131aa042fb.png)

 If the input is put right inside the Form and not passed as props.children, and the console.log(methods.watch()), also put inside the form component, it works fine. But i am trying to create some generic code.

Pardon my English, and i if that is a better way to do this i would love to know.

**Comment by bluebill1049** [comment, marked as answer]
this indicates your component is not been registered correctly.

<img width="762" alt="Screen Shot 2021-12-29 at 9 14 28 am" src="https://user-images.githubusercontent.com/10513364/147610969-ca363405-8da8-445c-b0c5-89abe075b079.png">

refer to your input, your `onChange` is overwrite react hook form's provided `onChange`
 
<img width="845" alt="Screen Shot 2021-12-29 at 9 15 32 am" src="https://user-images.githubusercontent.com/10513364/147611023-ef774dd8-580c-4f62-8f1b-ad3eff315626.png">

please send some time on the doc: https://react-hook-form.com/api/useform/register a lot of valuable information there.

**Reply by caiofelixuc** [reply]
Thanks Bill, iam a fan of your work! Shout out to Brazil! 🇧🇷

**Reply by bluebill1049** [reply]
Thank you @caiofelixuc! You are most welcome. ❤️ 🇧🇷
